### ISSUE TITLE
clang  link issue

### ISSUE THREAD

**Issue by dcoeurjo** [holzschu/a-shell#44]
while trying to compile some simple c++ code (single header, only using std::vector and iostream), I'm having strange link issues when running clang++:

```
wasm-ld: error: /private/var/mobile/Containers/Data/Application/F31CCAE9-535B-4BEC-866E-92B42B73A6B4/tmp/e
xample-b7e8bf.o: undefined symbol: __cxa_allocate_exception
wasm-ld: error: /private/var/mobile/Containers/Data/Application/F31CCAE9-535B-4BEC-866E-92B42B73A6B4/tmp/e
xample-b7e8bf.o: undefined symbol: __cxa_throw
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```
The code is this one (low discrepancy BN sampling, single header with a tiny main.cpp): https://github.com/dcoeurjo/LowDiscBlueNoise 

any idea ? 

(and thanks, Nicolas, for this very nice app!)

**Comment by holzschu** [comment]
Yes, it is part of the "Notable exceptions" of wasi-libc: https://github.com/WebAssembly/wasi-sdk#notable-limitations : 
"This repository does not yet support C++ exceptions. C++ code is supported only with -fno-exceptions for now."
I should maybe add this to the default flags.

**Comment by dcoeurjo** [comment]
🎉 🎉 🎉 🎉 
many thanks !

**Comment by dcoeurjo** [comment]
WASM build is super fast :)

**Comment by dcoeurjo** [comment]
Blue noise sampling on iPad;)

![IMG_5CB6A4D0621E-1](https://user-images.githubusercontent.com/700165/80384572-57aa9580-88a5-11ea-8b81-bebaac1e4604.jpeg)

**Comment by holzschu** [comment]
Wasm is a bit slower than native code in my tests, but not by much (around 40% slower, sometimes faster if you compile with "clang -O3" on the iPad).
